src/components/TextInput/v1/TextInput.md

```diff
@@ -0,0 +1,294 @@
+ - `rui_inputIconColor_error`
+ - `rui_inputIconColor_success`
+ - `rui_inputIconFontSize`
+ - `rui_inputIconPadding`
+ - `rui_inputIconRight`
+ - `rui_inputIconTop`
+ - `rui_inputIconTextPadding`
+
+##### Single line, text input.
+Default text input used in the majoprity of instances. It can be defined as one of four input types `"text"`, `"email"`, `"password"`,  `"url"`.
+```jsx
+<div style={{ display: "flex" }}>
+  <div style={{ border: "1px solid #cccccc", flex: "0 1 50%", padding: "1rem" }}>
+    <TextInput name="example" placeholder="Hint" />
+  </div>
+</div>
+``` 
+###### Input styles
+Text input on white and text input on grey or dark backgrounds. The dark brackground style is applyed by adding the `isDark` prop to the component. Use `dark` text input when the background is grey or dark. When a grey background is used to create hierarchy and sections a white text field is use. 
+```jsx
+<div style={{ display: "flex" }}>
+  <div style={{ border: "1px solid #cccccc", flex: "1 1 auto", padding: "1rem" }}>
+    <TextInput name="example" placeholder="Hint" />
+  </div>
+  <div style={{ backgroundColor: "#f5f5f5", flex: "1 1 auto", margin: "0 1rem", padding: "1rem" }}>
+    <TextInput name="example" placeholder="Hint" isDark />
+  </div>
+</div>
+```
+
+###### Input states
+Text input's have 8 basic states.
+ - Idle unfilled
+ - Idle filled
+ - Focused unfilled
+ - Focused filled
+ - Saved<sup>*</sup> *not yet implamented*
+ - Valid
+ - Invalid
+ - Read Only / Disabled
+
+Idle Unfilled / Focused Unfilled
+
+When a user has clicked or pressed into an input it’s in focused state. 
+```jsx
+<div style={{ display: "flex" }}>
+  <div style={{ border: "1px solid #cccccc", flex: "1 1 auto", padding: "1rem" }}>
+    <TextInput name="example" placeholder="Hint" />
+  </div>
+  <div style={{ backgroundColor: "#f5f5f5", flex: "1 1 auto", margin: "0 1rem", padding: "1rem" }}>
+    <TextInput name="example" placeholder="Hint" isDark />
+  </div>
+</div>
+```
+
+Idle Filled / Focused Filled
+
+When a field has been filled in by the user, and has been unfocused it shows idle and filled in. When a user has filled in a text input and clicked back into the text field it is shown as focused and has an icon button that allows the user to quickly clear the whole field and edit it from scratch.
+```jsx
+<div style={{ display: "flex" }}>
+  <div style={{ border: "1px solid #cccccc", flex: "1 1 auto", padding: "1rem" }}>
+    <TextInput name="example" placeholder="Hint" value="value" />
+  </div>
+  <div style={{ backgroundColor: "#f5f5f5", flex: "1 1 auto", margin: "0 1rem", padding: "1rem" }}>
+    <TextInput name="example" placeholder="Hint" value="value" isDark />
+  </div>
+</div>
+```
+
+Valid
+
+When information is required and/or needs to be formatted in a specific way the text input should be validated. When it’s successful the field shows a success state. Passing the `hasBeenValidated` prop to the input will enabled the valid state.
```

Comment:
Let's be careful to say "input" rather than "field" everywhere since we'll have another one called Field.

Also, clarify the last line here:

The "valid" state appearance is used when the `errors` prop is empty and the `hasBeenValidated` prop is true. Typically an ancestor form component will do the work of setting these properly for you.